TemporaryParser-Tests/TPRBScannerTest.class.st

```diff
@@ -1,146 +1,593 @@
+	| token tokens |
+	tokens := {TPRBBinarySelectorToken. TPRBErrorToken. TPRBIdentifierToken. TPRBKeywordToken.        	 	           TPRBLiteralToken. TPRBMultiKeywordLiteralToken. TPRBNumberLiteralToken. 	   	  	 	  	   	 				  TPRBPatternBlockToken. TPRBSpecialCharacterToken.}.
+	"Test the creation of a literal for every binary."
+	tokens do: [:tok | 
+		token := (self buildScannerForText: '!') scanBinary: tok.
+					 self assert: token class equals: tok.
+					 self assert: token value equals: '!'].
+]
+
+{ #category : #'scanBinary:' }
+TPRBScannerTest >> testScanBinaryWithLitteralToken [
+"This test is aimed at verifying the production a litteral token with the right properties when 	scanning binary character.
+	
+	binaryCharacter ::= '!' | '%' | '&' | '*' | '+' | ',' | '/' | '<' | '=' | '>' | '?' | '@' | '\' | 							 							 '~' | '|' | '-'"
+	| token binary |
+	binary := #('!' '%' '&' '*' '+' ',' '/' '<' '=' '>' '?' '@' '\' '~' '|' '-').
+	"Test the creation of a literal for every binary."
+	binary do: [:sbin | token := (self buildScannerForText: sbin) scanBinary: TPRBLiteralToken.
+							  self assert: token isLiteralToken.
+							  self assert: token value equals: sbin].
+]
+
+{ #category : #'scanBinary:' }
+TPRBScannerTest >> testScanBinaryWithLitteralTokenWithNonBinaryCharacterForFirstValue [
+	"This test is aimed at verifying the production a litteral token with the right properties when 	scanning non binary character as first character.
+	
+	binaryCharacter ::= '!' | '%' | '&' | '*' | '+' | ',' | '/' | '<' | '=' | '>' | '?' | '@' | '\' | 							 							 '~' | '|' | '-'"
+	| token nonbinary |
+	nonbinary := #('1' 'p' '^' 'é' '1=' '1=+').
+	"Test the creation of a literal for every binary."
+	nonbinary do: [:snbin | token := (self buildScannerForText: snbin) scanBinary: TPRBLiteralToken.
+							  self assert: token isLiteralToken.
+							  self assert: token value equals: snbin].
+]
+
+{ #category : #'scanBinary:' }
+TPRBScannerTest >> testScanBinaryWithLitteralTokenWithNonBinaryValues [
+"This test is aimed at verifying the production a litteral token with the right properties when 	scanning binary character.
+	
+	binaryCharacter ::= '!' | '%' | '&' | '*' | '+' | ',' | '/' | '<' | '=' | '>' | '?' | '@' | '\' | 							 							 '~' | '|' | '-'"
+	| token nonbinary |
+	nonbinary := #(#('1' '1') #('p' 'p') #('^' '^') #('1alpha' '1') #('11' '1') #('a b' 'a')).
+	"Test the creation of a literal for every binary."
+	nonbinary do: [:snbin | token := (self buildScannerForText: (snbin first)) scanBinary: 																											 					                                                                     TPRBLiteralToken.
+							  self assert: token isLiteralToken.
+							  self assert: token value equals: (snbin last)].
+]
+
+{ #category : #'scanError:' }
+TPRBScannerTest >> testScanErrorWithMultipleCharacters [

```

Comment:
We should look at users of scanError: in the scanner and deduce test cases from that. For example we say that

```smalltalk
scanLiteralCharacter
	| token |
	self step.	"$"
	currentCharacter ifNil:[ ^ self scanError:'A Character was expected'].
	token := RBLiteralToken 
				value: currentCharacter
				start: tokenStart
				stop: stream position.
	self step.	"char"
	^token

```

Then we can see that

```smalltalk
sc := RBScanner on: (ReadStream on: '#').
sc next.
```
=> an error token